packages/block-library/src/table-of-contents/hooks.js

```diff
@@ -106,63 +92,60 @@ function getLatestHeadings( select, clientId ) {
 					continue;
 				}
 
-				const canBeLinked =
-					typeof headingPageLink === 'string' &&
+				// Get the heading content
+				const content = stripHTML(
+					headingAttributes.content.replace( /(<br *\/?>)+/g, ' ' )
+				);
+
+				// Use explicit anchor or generate one from content
+				let anchor =
 					typeof headingAttributes.anchor === 'string' &&
-					headingAttributes.anchor !== '';
+					headingAttributes.anchor !== ''
+						? headingAttributes.anchor
+						: null;
+
+				// Auto-generate anchor from content if not set
+				if ( ! anchor && content ) {
+					anchor = content
+						.toLowerCase()
+						.replace( /[^\w\s-]/g, '' )
+						.replace( /[\s_]+/g, '-' )
+						.replace( /^-+|-+$/g, '' );
+				}
+
+				// Get the full permalink for consistency with server-side rendering.
+				// eslint-disable-next-line @wordpress/data-no-store-string-literals
+				const { getCurrentPostId, getCurrentPostType } =
```

Comment:
```suggestion
				// Get the full permalink for consistency with server-side rendering.
				const { getCurrentPostId, getCurrentPostType } =
```

Remove unnecessary eslint comment


Question: In a template context, does this permalink generation logic work correctly?

Reply:
I don't think we'll be able to get the ToC working this way in template context. With the current approach, I think folks will get stale values when it's placed this way.

May we should limit the block to `post-content` and note that it just doesn't work in template context yet.